gensim/similarity_learning/sl_vocab.py

```diff
@@ -0,0 +1,557 @@
+            self.word_counter.update(sentence.split())
+        for i, word in enumerate(self.word_counter.keys()):
+            self.word2index[word] = i
+            self.index2word[i] = word
+        self.vocab_size = len(self.word2index)
+        logger.info("Vocab Build Complete")
+
+        # Since the embedding layer will a different order, the indexes
+        # of the words might change. So, we'll make a revised word2index copy
+
+        revised_word2index = {}
+        revised_index2word = {}
+
+        logger.info("Building embedding index using pretrained word embeddings")
+        embeddings_index = {}
+        with open(self.word_embedding_path) as f:
+            for line in f:
+                values = line.split()
+                word = values[0]
+                coefs = np.asarray(values[1:], dtype='float32')
+                embeddings_index[word] = coefs
+
+        self.embedding_matrix = np.zeros(
+            (len(self.word2index) + 1, self.embedding_dim))
+        for word, i in self.word2index.items():
+            embedding_vector = embeddings_index.get(word)
+            if embedding_vector is not None:
+                # words not found in embedding index will be all-zeros.
+                self.embedding_matrix[i] = embedding_vector
+                revised_word2index[word] = i
+                revised_index2word[i] = word
+
+        # TODO check below, throwing slight error
+        # revise the value of index2word and word2index
+        # self.word2index = revised_word2index
+        # self.index2word = revised_index2word
+
+        logger.info("Embedding index build complete")
+
+    def preprocess(self, sentence):
+        """Preprocess an input string to allow only alphabets and numbers
+
+        Parameters:
+        ----------
+        sentence: str
+            The sentence to preprocess
+        """
+        return re.sub("[^a-zA-Z0-9]", " ", sentence.lower())
+
+    def make_indexed(self, sentence):
+        """Returns the indexed version of the sentence based on self.word2index
+        in the form of a list
+
+        Parameters:
+        -----------
+        sentenceL str
+            The sentence to be indexed
+        """
+        return [self.word2index[word] for word in sentence.split()]
+
+    def get_data(self, filter_queries=True):
+        """Provides the data in the format:
+        (query, document, relevance)
+        where query: list of ints representing the words in the query sentence
+              document: list of ints representing the words in the document sentence
+              relevance: the relevance score of the two in terms of 1 or 0
+
+        Parameters:
+        ----------
+        filter_queries: boolean
+            Whether queries which have no relevant answer should be kept
+            if True, no answer queries are dropped
+        """
+
+        self.data = []
+        for Question, Answer in self.df.groupby('QuestionID').apply(dict).items():
```

Comment:
As I understand, this will be part of the codebase in future, for this reason, remove pandas everywhere please